pyntc/devices/ios_device.py

```diff
@@ -541,12 +543,53 @@ def file_copy_remote_exists(self, src, dest=None, file_system=None):
             return True
         return False
 
-    def install_os(self, image_name, **vendor_specifics):
+    def install_os(self, image_name, install_mode=False, **vendor_specifics):
+        """Installs the prescribed Network OS, which must be present before issuing this command.
+
+        Args:
+            image_name (str): Name of the IOS image to boot into
+            install_mode (bool, optional): Uses newer install method on devices. Defaults to False.
+
+        Raises:
+            OSInstallError: Unable to install OS Error type
+
+        Returns:
+            bool: False if no install is needed, true if the install completes successfully
+        """
         timeout = vendor_specifics.get("timeout", 3600)
         if not self._image_booted(image_name):
-            self.set_boot_options(image_name, **vendor_specifics)
-            self.reboot(confirm=True)
+            if install_mode:
+                # Change boot statement to be boot system <flash>:packages.conf
+                self.set_boot_options(INSTALL_MODE_FILE_NAME, **vendor_specifics)
+                delay_factor = 10
```

Comment:
Could the delay factor be included as another constant or even a positional argument? It may add some flexibility in fine tuning it.

Reply:
👍